### ISSUE TITLE
Uncaught Error: ENOENT, common/api/lib/shell.js not found in /usr/share/atom/resources/atom.asar

### ISSUE THREAD

**Issue by fdinardo** [Gert-dev/php-integrator-navigation#16]
Steps to reproduce:

1. Install php-integrator-navigation
 - or 
1. Disable php-integrator-navigation 
2. Enable  php-integrator-navigation 

Attempts:
I tried to change the key for navigation in the settings, but it still happen.

**Atom Version**: 1.7.0
**System**: linux 4.4.5-1-ARCH
**Thrown From**: [php-integrator-navigation](https://github.com/Gert-dev/php-integrator-navigation) package, v0.6.2

### Stack Trace

Uncaught Error: ENOENT, common/api/lib/shell.js not found in /usr/share/atom/resources/atom.asar

```
At ATOM_SHELL_ASAR.js:114

Error: ENOENT, common/api/lib/shell.js not found in /usr/share/atom/resources/atom.asar
    at notFoundError (ATOM_SHELL_ASAR.js:110:13)
    at Object.fs.readFileSync (ATOM_SHELL_ASAR.js:435:9)
    at compileFileAtPath (/usr/share/atom/resources/app.asar/src/compile-cache.js:69:23)
    at Object.defineProperty.value [as .js] (/usr/share/atom/resources/app.asar/src/compile-cache.js:207:18)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:313:12)
    at Module.require (module.js:366:17)
    at require (/usr/share/atom/resources/app.asar/src/native-compile-cache.js:50:27)
    at Object.<anonymous> (/home/angelsoul/.atom/packages/php-integrator-navigation/lib/ClassProvider.coffee:2:9)
    at Object.<anonymous> (/home/angelsoul/.atom/packages/php-integrator-navigation/lib/ClassProvider.coffee:1:1)
    at Module._compile (/usr/share/atom/resources/app.asar/src/native-compile-cache.js:103:30)
    at Object.defineProperty.value [as .coffee] (/usr/share/atom/resources/app.asar/src/compile-cache.js:208:21)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:313:12)
    at Module.require (module.js:366:17)
    at require (/usr/share/atom/resources/app.asar/src/native-compile-cache.js:50:27)
    at Object.module.exports.activateProviders (/home/angelsoul/.atom/packages/php-integrator-navigation/lib/Main.coffee:64:33)
    at Object.module.exports.setService (/home/angelsoul/.atom/packages/php-integrator-navigation/lib/Main.coffee:99:10)
    at Provider.module.exports.Provider.provide (/usr/share/atom/resources/app.asar/node_modules/service-hub/lib/provider.js:30:52)
    at /usr/share/atom/resources/app.asar/node_modules/service-hub/lib/service-hub.js:55:26
    at doNTCallback0 (node.js:443:9)
    at process._tickCallback (node.js:372:13)
```

### Commands

```
```

### Config

```json
{
  "core": {
    "autoHideMenuBar": true,
    "disabledPackages": [
      "you-complete-me",
      "linter-bootlint",
      "atom-beautify",
      "autocomplete-clang",
      "autocomplete-python",
      "linter-mcs",
      "emmet",
      "editorconfig",
      "autoclose-html",
      "symbols-view"
    ],
    "excludeVcsIgnoredPaths": false,
    "packagesWithKeymapsDisabled": [
      "markdown-preview"
    ],
    "projectHome": "/home/angelsoul/myWorks",
    "themes": [
      "atom-material-ui",
      "atom-material-syntax"
    ]
  },
  "php-integrator-navigation": {}
}
```

### Installed Packages

```coffee
# User
atom-alignment, v0.12.1
atom-material-syntax, v0.4.6
atom-material-ui, v1.2.10
atom-minify, v0.7.1
atom-ternjs, v0.13.2
autocomplete-paths, v1.0.2
caniuse, v0.9.0
color-picker, v2.1.1
docblockr, v0.7.3
file-icons, v1.7.1
fonts, v1.2.0
highlight-line, v0.11.1
highlight-selected, v0.11.2
jquery-snippets, v11.0.0
language-blade, v0.24.1
language-cshtml, v0.1.1
laravel, v0.7.0
linter, v1.11.4
linter-csslint, v1.3.2
linter-htmlhint, v1.1.2
linter-jshint, v2.1.0
linter-php, v1.2.0
linter-pylint, v1.2.0
markdown-scroll-sync, v2.1.2
minimap, v4.21.0
minimap-git-diff, v4.2.0
minimap-highlight-selected, v4.4.0
minimap-pigments, v0.2.0
php-debug, v0.2.2
php-integrator-autocomplete-plus, v0.7.0
php-integrator-base, v0.8.0
php-integrator-call-tips, v0.2.3
php-integrator-linter, v0.2.2
php-integrator-navigation, v0.6.2
php-integrator-refactoring, v0.4.2
php-integrator-tooltips, v0.6.0
pigments, v0.26.0
rest-client, v1.0.0
symbol-gen, v1.0.0
tab-control, v0.6.9
tree-view-breadcrumb, v0.8.1

# Dev
No dev packages
```
I don't know if this is only my own problem (related to my bad configuration of PHP or Atom itself), others php-integrator-* packages works fine.

If I try to uninstall the package, Atom says:
```
Failed to delete php-integrator-navigation: Does not exist
```
but it is installed...

**Comment by Gert-dev** [comment]
Hello. This sounds like an Atom bug or an incorrect Atom build, it seems the php-integrator-navigation package is being loaded and Atom is requested to compile the CoffeeScript files, but it then suddenly errors out. The referenced file `/usr/share/atom/resources/atom.asar` is a file from Atom itself that is incorrect. Perhaps they broke something in the new release.

I can't reproduce it right now, but I'll try again once I upgrade to 1.7.0.

**Comment by fdinardo** [comment]
Hello. Thank you for the fast response :D. You did a GREAT job with these php-integrator packages!

**Comment by Gert-dev** [comment]
I've updated to 1.7.0 and it seems to be an Arch Linux problem:
* atom-editor has patches for 1.7.0 because the version of freetype2 shipping with Arch Linux breaks with the Electron release shipped with Atom 1.7.0. They bumped the Electron release to fix the issue, but it seems they've caused other problems by doing it.
* atom-editor-bin is still at 1.6.2, but if you manually update to the 1.7.0 binary, you will get a freeze at startup due to the freetype2 package problem. You can fix this by downgrading the freetype2 package.

See also [1] and [2]. I'll close this as it seems to be an Atom issue, there isn't much I can do about it, sorry :(.

[1] https://github.com/atom/atom/issues/11198
[2] https://github.com/atom/atom/issues/11457
